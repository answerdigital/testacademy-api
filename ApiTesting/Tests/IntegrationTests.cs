namespace ApiTesting.Tests
{
    using System;
    using static RestAssured.Dsl;
    using FluentAssertions;
    using Microsoft.AspNetCore.Builder;
    using Microsoft.Extensions.Hosting;
    using Microsoft.AspNetCore.Mvc.Testing;
    using Microsoft.AspNetCore.Hosting;
    using Dotnet.AspNetCore.Samples.WebApi;
    using System.Security.AccessControl;
    using System.Runtime.CompilerServices;
    using ApiTesting.Classes;
    using Newtonsoft.Json.Linq;
    using Snapshooter.NUnit;
    using RestAssured.Response;
    using ApiTesting.Classes.PostData;

    public class IntegrationTests : BasicTests
    {

        [TestCase("Players")]
        [TestCase("Players/1")]
        public void StatusCodeIndicatingSuccessCanBeVerifedForGetRequests(string url)
        {
            Given(httpClient)
                .When()
                .Get(baseUrl + url)
                .Then()
                .StatusCode(200);
        }

        [TestCase("Players")]
        [TestCase("Players/1")]
        public void ResponseContentTypeHeaderCanBeVerifiedForGetRequests(string url)
        {
            Given(httpClient)
            .When()
            .Get(baseUrl + url)
            .Then()
            .StatusCode(200)
            .ContentType("application/json; charset=utf-8");
        }

        [Test]
        public void JsonStringResponseBodyCanBeVerifiedForGetRequests()
        {
            JObject testPlayer = (JObject)Given(httpClient)
            .When()
            .Get(baseUrl + "Players/1")
            .Then()
            .StatusCode(200)
            .DeserializeTo(typeof(JObject));

            Snapshot.Match(testPlayer);
        }

        [Test]
        public void Responsebodylengthcanbeverified()
        {
            List<object> players = (List<object>)Given(httpClient)
            .When()
            .Get(baseUrl + "Players")
            .Then().StatusCode(200).Extract().Body("$.[0:]");

            Assert.That(players.Count, Is.EqualTo(11));

        }

        [Test]
        public void ResponseTimeCanBeVerified()
        {
            Given(httpClient)
            .When()
            .Get(baseUrl + "Players")
            .Then()
            .ResponseTime(NHamcrest.Is.LessThan(TimeSpan.FromMilliseconds(200)));
        }

        [Test]
        public void SucessStatusCodeCanBeVerifiedForPostRequest()
        {
            DataHelper dataHelper = new DataHelper();

            Player player = dataHelper.GetPlayerData("Post_Harvey");

         object orderResponse = Given(httpClient)
        .Body(player)
        .When()
        .Post(baseUrl + "Players")
        .Then()
        .StatusCode(201)
        .Extract().Body("$");

         Snapshot.Match(orderResponse, matchOptions => matchOptions.IgnoreField("id"));
        }
    }
}

