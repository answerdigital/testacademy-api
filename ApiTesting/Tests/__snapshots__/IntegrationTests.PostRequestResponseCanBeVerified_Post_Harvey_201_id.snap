{
  "id": 13,
  "firstName": "Harvey",
  "middleName": "The Power",
  "lastName": "Sembhy",
  "dateOfBirth": "1066-06-06T14:30:23.234Z",
  "squadNumber": 69,
  "position": "Everywhere",
  "abbrPosition": "EW",
  "team": "Ninja Warriors",
  "league": "Answer under 1100s",
  "starting11": false
}
